Sources/NIOHTTP1/HTTPServerPipelineHandler.swift

```diff
@@ -0,0 +1,106 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2017-2018 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+//
+//  HTTPServerPipelineHandler.swift
+//  NIOHTTP1
+//
+//  Created by Cory Benfield on 01/03/2018.
+//
+
+import NIO
+
+/// A `ChannelHandler` that handles HTTP pipelining by buffering inbound data until a
```

Comment:
@Lukasa maybe we want to add a link to: https://tools.ietf.org/html/rfc7230#section-6.3.2 just to add a bit more context ?